### PR TITLE
Improve disk checks

### PR DESCRIPTION
**Pull request by htcfreek**
# Details
This PR improves the disk checks: GPT and space. The PR introduces changes in UI and check/WMI code.
- Now show only result for system disk and system partition. (Where Windows directory of bootet OS is located).
- New checks can return an array with check data for all internal disks:
  - GPT: `["Disk number", "Partition style", "Check result"]`
  - Space: `["Disk number", "Size (GB)", "Check result"]`
- Information displayed to user is improved.

Additions:
- Update `_GetDiskInfoFromWmi()` to v1.4.1
- Small improvements and fixes on code base.

**Fixes:** #385, #400, #417

### Screenshots
wnw11-pr431_2
image

# Tested
- [x] Test different possible results of the checks.
- [x] Validated data shown in Gui.
- [x] Tested that other disks than the system disk not influences the checks results.
image

---

### ToDo (Open items):
- [x] Fix: Typos
- [x] Fix: `_GetDiskProperties(4)` doesn't return correct data.
- [x] Add feature to return "DiskResultTables"
- [x] Update lang files.
- [x] ⚠️ Fix merge conflicts.
- [x] Final check and tidy.

### CODE REVIEW

File: WhyNotWin11.au3 [hunk 1]
```diff
@@ -48,17 +50,18 @@ If @OSVersion = 'WIN_10' Then DllCall(@SystemDir & "\User32.dll", "bool", "SetPr
 #include <StringConstants.au3>
 #include <WindowsConstants.au3>
 
-Global $WINDOWS_DRIVE = EnvGet("SystemDrive")
-
+; Include external functions
 #include "Includes\ResourcesEx.au3"
+#include "Includes\GetDiskInfoFromWmi.au3"
 
+; Internals internal scripts
```
Comment (htcfreek) [inline WhyNotWin11.au3:57]: Type:

```suggestion
; Include internal scripts
```

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,6 @@ WhyNot.txt
 WhyNotWin11_stripped.au3
 del.au3
 Config/config.st
+
+# Backupfolder from Tidy in "ISN AutoIt Studio"
```
Comment (htcfreek) [inline .gitignore:7]: Typo:

```suggestion
# Backup directory from Tidy in "ISN AutoIt Studio"
```

**Review by htcfreek — COMMENTED** [review]
Note to me: Fix typos.

**Comment by htcfreek** [comment]
@rcmaehl 
If you want to show the "x/y internal disks meet requirements."-Note I need to know where to show it.

**Comment by htcfreek** [comment]
@micwoj92 
Which changes in lang file metadata should I do after updating them?

**Comment by micwoj92** [comment]
When you update, you should add your name to translators and bump version, so it comes over with latest dev version.

**Comment by htcfreek** [comment]
> @rcmaehl 
> If you want to show the "x/y internal disks meet requirements."-Note I need to know where to show it.

ping @rcmaehl

File: WhyNotWin11.au3 [hunk 3]
```diff
@@ -533,13 +534,22 @@ Func Main()
...
-	If _SpaceCheck() Then
-		_GUICtrlSetPass($hCheck[9][0])
-	Else
-		_GUICtrlSetFail($hCheck[9][0])
+	#Region - SpaceCheck
+	$sSpaceResultFirstLine = _SpaceCheck(3) & " " & _SpaceCheck(4) & " GB, " & _Translate($iMUI, "Disk") &  " " & _SpaceCheck(1) & ": " & _SpaceCheck(2) & " GB"
+	If _SpaceCheck(4) >=  64 And _SpaceCheck(2) >= 64 Then
+		; Partition and Disk >= 64 GB
+		_GUICtrlSetState($hCheck[9][0], $iPass)
+		GUICtrlSetData($hCheck[9][2], $sSpaceResultFirstLine & @CRLF & _Translate($iMUI, "Both passed."))
```
Comment (htcfreek) [inline WhyNotWin11.au3:542]: Partition C: (xyz GB): passed
Disk 0 (xyz GB): not passed

or

Disk: passed, Partition passed

**Review by htcfreek — COMMENTED** [review]
Ideas

**Comment by htcfreek** [comment]
@rcmaehl 
Please review this. If everything is fine you can fix the merge conflicts and merge this pr.

File: Langs/0000.lang [hunk 4]
```diff
@@ -39,14 +39,17 @@ Not Currently Listed as Compatible=
 Listed as Compatible=
 Cores=
 Threads=
-GPT Detected=
-GPT Not Detected=
+Requirement meet.=
```
Comment (micwoj92) [inline Langs/0000.lang:42]: Shouldnt it be "requirements met" in this situation?

File: WhyNotWin11.au3 [hunk 5]
```diff
@@ -540,13 +541,25 @@ Func Main()
...
+	#Region - SpaceCheck
+	Local $iDiskSpace = _SpaceCheck(2)
+	Local $iPartitionSpace = _SpaceCheck(4)
+	Local $sSpaceResultFirstLine = _SpaceCheck(3) & " " & $iPartitionSpace & " GB, " & _Translate($iMUI, "Disk") & " " & _SpaceCheck(1) & ": " & $iDiskSpace & " GB"
+	; -- Check
+	If $iDiskSpace >= 64 And $iPartitionSpace >= 64 Then
+		; Partition and Disk >= 64 GB
 		_GUICtrlSetState($hCheck[9][0], $iPass)
-	Else
+		GUICtrlSetData($hCheck[9][2], $sSpaceResultFirstLine & @CRLF & _Translate($iMUI, "Partition and disk passed."))
```
Comment (micwoj92) [inline WhyNotWin11.au3:552]: These strings also need to be added to translation file

Comment (htcfreek) [reply WhyNotWin11.au3:552]: Is there.

File: Langs/0000.lang (hunk 4 above)
Comment (htcfreek) [reply Langs/0000.lang:42]: No.

Requirement = Singular (one)
Requirements = Plural (two+)

I think we check many requiremen*ts*, but in this check only one requiremen*t* must be fulfilled.

Comment (rcmaehl) [reply Langs/0000.lang:42]: Met, not meet.

Comment (htcfreek) [reply Langs/0000.lang:42]: Ooops.

**Comment by htcfreek** [comment]
@rcmaehl 
Conflicts fixed; Typo fixed; Tidy source done. Let's merge 🚀

**Comment by micwoj92** [comment]
@htcfreek still has conflicts.

**Comment by htcfreek** [comment]
@rcmaehl, @micwoj92 
Can you please merge before we get new merge conflicts again. 🚀

**Comment by micwoj92** [comment]
Conflicts again

**Comment by htcfreek** [comment]
Now @rcmaehl  has to fix them I can't fix them for nothing every day.

**Comment by micwoj92** [comment]
My opinion is that it would be better if it was split into 4 or more smaller pull requests. Less conflicts and could be reviewed faster.

**Comment by htcfreek** [comment]
> My opinion is that it would be better if it was split into 4 or more smaller pull requests. Less conflicts and could be reviewed faster.

But this makes not really sense. How would you split it?

**Comment by htcfreek** [comment]
@rcmaehl 
When can you review this PR?

**Comment by rcmaehl** [comment]
Will be making changes after merge but okay

**Comment by rcmaehl** [comment]
I am seeing significantly worse check performance on my system compared to pre-merge. Please investigate and static variables where needed for WMIC calls.

Slow

**Comment by htcfreek** [comment]
@rcmaehl 
Do you mean the moment when progress shows "User Interface". At this point the wmi calls ate done. And I added static variables where I thought it makes sense.

**Comment by micwoj92** [comment]
> > My opinion is that it would be better if it was split into 4 or more smaller pull requests. Less conflicts and could be reviewed faster.
> 
> But this makes not really sense. How would you split it? These changes are all related to ech other and splitting them would create code braking PRs.
> 
> The only file which makes conflicts is the main script.

One for update to GetDiskInfoFromWmi 1.4.1, one to change text on checks and one to show only result for system disk. And you also listed "Small improvements and fixes on code base." So that also could be separate.

**Comment by micwoj92** [comment]
Oh, I see Robert updated contribution guidelines, seem reasonable to me.

**Pull request merged**